### PR TITLE
configurable noHtml5Validate prop for NextScaffolderPage

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

HTML5 validation is on by default in the `@rjsf/material-ui-v5` Form in `next/Stepper` because the `noHtml5Validate` prop is not set on the `<Form />` component.

The contrary is true in the `MultiJsonForm` where `noHtml5Validate` is true and only true.

This PR makes it configurable by adding a `noHtml5Validate` validate prop to the `NextScaffolderPage`.

I could set it to true on the form and remove the prop as an even simpler PR.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by dagda1** [comment]
There is a noticeable deterioration in the quality of validation messages between version `3.2.1` and `5.0.0-beta.14`.

Here is a [codesandbox](https://codesandbox.io/s/objective-forest-95mqp1?file=/src/App.js) of version 3.2.1 and a screen shot:

validations-3-2-1

This is great, and I do not have to do anything, and the validation works by default very nicely.  This is totally great :)

And here is a [codesandbox](https://codesandbox.io/p/github/dagda1/rjsf-example-5/draft/trusting-noyce?file=%2Fsrc%2FApp.tsx&selection=%5B%7B%22endColumn%22%3A30%2C%22endLineNumber%22%3A14%2C%22startColumn%22%3A30%2C%22startLineNumber%22%3A14%7D%5D) of v5 and a screenshot:

v5

The default error messages are unusable, they point to the field name, and the input is not highlighted. 

I would have to (if backstage even exposed it) use transformErrors in every single form, which is a waste.

It is either HTML5 validation or nothing in the new world.  I realise this is a `@rsjf` problem but I wonder if there is anything we can do as a workaround in this PR?

**Comment by benjdlambert** [comment]
I wonder if we can nest these props under something like `StepperProps` or something instead of polluting the props directly?

**Comment by benjdlambert** [comment]
Hmm, just looking through the `rjsf` playground, it looks like that maybe it could be something that we're doing wrong? Under the `Errors` tab, it looks like they have `ContextualizedErrors` that render in the right place? It could be that we're doing something funky with the `extraErrors` part and that they're not being rendered in the right place or something, might be worth digging in a little bit to see if we're doing something wrong there.

Screenshot 2022-12-19 at 10 50 12

File: plugins/scaffolder/api-report.md [hunk 1]
```diff
@@ -264,6 +264,7 @@ export type NextRouterProps = {
   };
   groups?: TemplateGroupFilter[];
   transformErrors?: ErrorTransformer;
+  noHtml5Validate?: boolean;
```
Comment (camilaibs) [inline plugins/scaffolder/api-report.md:267]: Is this something we would like for all templates? If not, have you considered adding this as a [template](https://backstage.io/docs/features/software-templates/adding-templates) optional configuration?

File: plugins/scaffolder/src/next/Router/Router.tsx [hunk 2]
```diff
@@ -46,6 +46,7 @@ export type NextRouterProps = {
   };
   groups?: TemplateGroupFilter[];
   transformErrors?: ErrorTransformer;
+  noHtml5Validate?: boolean;
```
Comment (camilaibs) [inline plugins/scaffolder/src/next/Router/Router.tsx:49]: We have documentation explaining how to customize [actions](https://backstage.io/docs/features/software-templates/writing-custom-actions), [fields](https://backstage.io/docs/features/software-templates/writing-custom-field-extensions) and [step](https://backstage.io/docs/features/software-templates/writing-custom-step-layouts) layouts. How about adding one for the form customization as well?

**Review by camilaibs — COMMENTED** [review]
Thanks for contributing, added minor suggestions 🙂

File: .changeset/eighty-pans-remain.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': patch
+---
+
+configurable noHtml5Validate prop for NextScaffolderPage
```
[the remark this answers is not in the document]
Comment (benjdlambert) [reply .changeset/eighty-pans-remain.md:5]:
```suggestion
Added `noHtml5Validate` prop to `FormProps` on `NextScaffolderPage`
```

[the remark this answers is not in the document]
Comment (benjdlambert) [reply .changeset/eighty-pans-remain.md:5]: Maybe an example too like @camilaibs mentioned too could be useful.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**